Sources/MongoSwift/BSON/BSONValue.swift

```diff
@@ -70,17 +70,16 @@ internal protocol BSONValue: Codable {
      *   - key: A `String`, the key under which to store the value.
      *
      * - Throws:
-     *   - `MongoError.InternalError` if the `DocumentStorage` would exceed the maximum size by encoding this
+     *   - `BSONError.InternalError` if the `BSONDocument` would exceed the maximum size by encoding this
```

Comment:
ditto updating this comment. I'd just do a check and make sure all appropriate docstrings are updated to reflect the new error type.

Reply:
I went around updating so that's all good but then I found:
https://github.com/mongodb/mongo-swift-driver/blob/cd45de9df57a3a71ec5ffe1da39d3355b1c56abe/Sources/MongoSwift/BSON/BSONDocument.swift#L209-L214
You think it makes sense to also make a TooSmall version too? or maybe go generic DocumentSizeError?

Reply:
I think this is fine as an internal error, since there really should be no way for a user to create a document that is smaller than 5 bytes.